### PR TITLE
Fix #537 - reduceRight()

### LINKED ISSUE #537: Fix reduceRight
There is currently a method called `reduceRight` that lives in `/lib/reduceRight.js`.
It is incomplete and needs to be fixed!

Like your usual reduce, but works from right to left.

### Acceptance Criteria:
1. Running `yarn test <methodName>` results in tests passing.
2. You have written a skeleton method for someone else to work on.
3. You have written tests surrounding your skeleton method.
4. Running `yarn lint` does not print any errors to the console!
5. Optional: write one or two more tests surrounding the method you are solving to account for potential edge cases.

### Please include the skeleton of a new method + an accompanying test for someone else to work on at the time of creating a pull request! A pull request will most likely be denied if it does not contain a skeleton method for someone else to work on! For more information, please read the [Contributing Guide](https://github.com/BlakeGuilloud/ganon/blob/master/CONTRIBUTING.md).

Thank you so much for your contribution!

### PR DESCRIPTION
**Pull request by rgehan**
Closes #537

- [] Running `yarn lint` does not trigger any linter errors
- [] The test of the method you have fixed is passing
- [] You have written a new skeleton method for someone else to work on!
- [] You have written tests to accompany your skeleton method

### CODE REVIEW

File: lib/reduceRight.js [hunk 1]
```diff
@@ -7,7 +7,17 @@
...
-  throw new Error("Still a skeleton");
+  if (typeof iteratee !== "function") {
+    throw "Iteratee should be a function";
+  }
+
+  if (!Array.isArray(list)) {
+    throw "List should be an array";
+  }
+
+  const item = list.pop();
```
Comment (SamHowie) [inline lib/reduceRight.js]: This mutates the bound list parameter.

Is this side effect expected?
